ria_remote/export_archive.py

```diff
@@ -0,0 +1,174 @@
+# emacs: -*- mode: python; py-indent-offset: 4; tab-width: 4; indent-tabs-mode: nil -*-
+# ex: set sts=4 ts=4 sw=4 noet:
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the datalad package for the
+#   copyright and license terms.
+#
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""Report status of a dataset (hierarchy)'s work tree"""
+
+__docformat__ = 'restructuredtext'
+
+
+import logging
+import os
+import os.path as op
+from hashlib import md5
+import subprocess
+from argparse import REMAINDER
+
+from datalad.utils import (
+    rmtree,
+)
+from datalad.interface.base import (
+    Interface,
+    build_doc,
+)
+from datalad.interface.results import (
+    get_status_dict,
+)
+from datalad.interface.utils import eval_results
+from datalad.support.param import Parameter
+from datalad.support.constraints import (
+    EnsureNone,
+    EnsureStr,
+)
+from datalad.distribution.dataset import (
+    EnsureDataset,
+    datasetmethod,
+    require_dataset,
+    resolve_path,
+)
+from datalad.log import log_progress
+from datalad.dochelpers import (
+    exc_str,
+)
+
+lgr = logging.getLogger('ria_remote.export_archive')
+
+
+@build_doc
+class ExportArchive(Interface):
+    """Export an archive of a local annex object store for the RIA remote.
+    """
+    # make the custom renderer the default one, as the global default renderer
+    # does not yield meaningful output for this command
+    _params_ = dict(
+        dataset=Parameter(
+            args=("-d", "--dataset"),
+            doc="""specify the dataset to process.  If
+            no dataset is given, an attempt is made to identify the dataset
+            based on the current working directory""",
+            constraints=EnsureDataset() | EnsureNone()),
+        outputdir=Parameter(
+            args=("outputdir",),
+            metavar="DIRECTORY",
+            doc="""directory to place the archive into""",
+            constraints=EnsureStr() | EnsureNone()),
+        opts=Parameter(
+            args=("opts",),
+            nargs=REMAINDER,
+            metavar="OPTIONS",
+            doc="""additional options for 7z"""),
```

Comment:
"additional" is somewhat misleading, since it replaces  that `-mx0` option.

Reply:
True, will fix.